tst/EnergyPlus/unit/Coils/CoilCoolingDX.unit.cc

```diff
@@ -64,3 +73,1591 @@ TEST_F( CoilCoolingDXTest, CoilCoolingDXInput )
+    evapInletNode.MassFlowRate = thisCoil.performance.normalMode.speeds.front().RatedAirMassFlowRate;
+    int useAlternateMode = DataHVACGlobals::coilNormalMode;
+    Real64 PLR = 1.0;
+    int speedNum = 1;
+    Real64 speedRatio = 1.0;
+    int fanOpMode = 1;
+    bool singleMode = false;
+    thisCoil.simulate(*state, useAlternateMode, PLR, speedNum, speedRatio, fanOpMode, singleMode);
+    if (!setExpectations) {
+        std::cout << thisCoil.totalCoolingEnergyRate << ',' << evapOutletNode.Temp << ',' << evapOutletNode.HumRat << std::endl;
+    } else {
+        EXPECT_NEAR(5000, thisCoil.totalCoolingEnergyRate, 0.1); // expect the coil to run full out, at speed 1
+        EXPECT_NEAR(10.238, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(0.007748, evapOutletNode.HumRat, 0.0001);
+    }
+    // alter values and run at rated conditions normal mode speed 2
+    evapInletNode.MassFlowRate = thisCoil.performance.normalMode.speeds.back().RatedAirMassFlowRate;
+    speedNum = 2;
+    thisCoil.simulate(*state, useAlternateMode, PLR, speedNum, speedRatio, fanOpMode, singleMode);
+    if (!setExpectations) {
+        std::cout << thisCoil.totalCoolingEnergyRate << ',' << evapOutletNode.Temp << ',' << evapOutletNode.HumRat << std::endl;
+    } else {
+        EXPECT_NEAR(10000, thisCoil.totalCoolingEnergyRate, 0.01); // expect the coil to run full out, at speed 1
+        EXPECT_NEAR(10.247, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(0.00774, evapOutletNode.HumRat, 0.0001);
+    }
+
+    // ok so now run at alternate mode, speed 1
+    useAlternateMode = DataHVACGlobals::coilEnhancedMode;
+    speedNum = 1;
+    thisCoil.simulate(*state, useAlternateMode, PLR, speedNum, speedRatio, fanOpMode, singleMode);
+    if (!setExpectations) {
+        std::cout << thisCoil.totalCoolingEnergyRate << ',' << evapOutletNode.Temp << ',' << evapOutletNode.HumRat << std::endl;
+    } else {
+        EXPECT_NEAR(4500, thisCoil.totalCoolingEnergyRate, 0.01); // expect the coil to run full out, at speed 1
+        // EXPECT_NEAR(20.373, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(20.411, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(0.0111, evapOutletNode.HumRat, 0.0001);
+    }
+
+    // ok so now run at alternate mode, speed 2
+    useAlternateMode = DataHVACGlobals::coilEnhancedMode;
+    speedNum = 2;
+    thisCoil.simulate(*state, useAlternateMode, PLR, speedNum, speedRatio, fanOpMode, singleMode);
+    if (!setExpectations) {
+        std::cout << thisCoil.totalCoolingEnergyRate << ',' << evapOutletNode.Temp << ',' << evapOutletNode.HumRat << std::endl;
+    } else {
+        EXPECT_NEAR(9000, thisCoil.totalCoolingEnergyRate, 0.01); // expect the coil to run full out, at speed 1
+        // EXPECT_NEAR(12.163, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(12.239, evapOutletNode.Temp, 0.01);
+        EXPECT_NEAR(0.00833, evapOutletNode.HumRat, 0.0001);
+    }
+}
+
+TEST_F(EnergyPlusFixture, DISABLED_CoilDXCoolingVsMultiSpeed_CycFanCycCoil)
```

Comment:
New unit tests to compare new coil vs old coil in a unit test. Extremely helpful, but these fail here without the old coil dx fixes in #8576 so disabled here.